worker/httpserverargs/manifold.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package httpserverargs
+
+import (
+	"reflect"
+
+	"github.com/juju/errors"
+	"github.com/juju/utils/clock"
+	"gopkg.in/juju/worker.v1"
+	"gopkg.in/juju/worker.v1/dependency"
+	"gopkg.in/tomb.v2"
+
+	"github.com/juju/juju/apiserver/apiserverhttp"
+	"github.com/juju/juju/apiserver/httpcontext"
+	workerstate "github.com/juju/juju/worker/state"
+)
+
+// ManifoldConfig holds the resources needed to run an httpserverargs
+// worker.
+type ManifoldConfig struct {
+	ClockName string
```

Comment:
I don't expect you to change this, but I think a manifold that just holds the clock is a waste given that we always need it and the original manifold config has the clock. I prefer to just pass the clock itself in rather than get it from the clock worker. I'd loke to change all workers to do this.